ick/base_rule.py

```diff
@@ -105,9 +223,42 @@ def list(self) -> ListResponse:
             rule_names=[self.rule_config.name],
         )
 
-    def prepare(self) -> None:
-        raise NotImplementedError(self.__class__)
+    def prepare(self) -> bool:
+        return True  # no setup required
 
-    @contextmanager
-    def work_on_project(self, project_path: Path):  # type: ignore[no-untyped-def] # FIX ME
-        yield self.work_cls(self, project_path)
+    def add_steps_to_run(self, projects: Any, run: Run) -> None:
+        qualname = self.rule_config.qualname
+
+        if self.rule_config.scope == Scope.FILE:
+            for p in projects:
+                run.add_step(
+                    GenericPreparedStep(
```

Comment:
I love this command (especially compared to the previous part that checked scope) -- it encapsulates a command to run and the patterns we care about, and we don't have to call `prepare` early anymore.